interpreter/valid/valid.ml

```diff
@@ -96,13 +96,13 @@ let refer_func (c : context) x = refer "function" c.refs.Free.funcs x
 
 let cont_type_of_heap_type (c : context) (ht : heap_type) at : cont_type =
   match ht with
-  | DefHT dt -> assert false
+  | DefHT dt -> as_cont_str_type (expand_def_type dt) (* TODO(dhil): we should always hit VarHT *)
```

Comment:
@rossberg please take a look here. I hit this case whenever a continuation type annotation is a recursive group. Recalling our previous discussions, I believe it should always be the case that defined heap types have become indexed heap types.

Reply:
Yeah, checking through the definitions, DefHT is indeed introduced when unrolling a recursive type index (in types.ml:unroll_rec_type), for which there generally is no (locally) known regular type index. So this can legally come up. Make sure we have tests for that.

Reply:
Gotcha! Thanks.